app/task/project/loadotio/loadotio.cpp

```diff
@@ -88,11 +89,36 @@ bool LoadOTIOTask::Run()
 
   // Keep track of imported footage
   QMap<QString, Footage*> imported_footage;
+  QMap<OTIO::Timeline*, Sequence*> timeline_sequnce_map;
 
+  // Variables used for loading bar
+  float number_of_clips = 0;
+  float clips_done = 0;
+
+  // Generate a list of sequences with the same names as the timelines.
+  // Assumes each timeline has a unique name.
   foreach (auto timeline, timelines) {
-    // Create sequence
     Sequence* sequence = new Sequence();
     sequence->SetLabel(QString::fromStdString(timeline->name()));
+    // Set default params incase they aren't edited.
+    sequence->set_default_parameters();
+    timeline_sequnce_map.insert(timeline, sequence);
+
+    // Get number of clips for loading bar
+    foreach (auto track, timeline->tracks()->children()) {
+      auto otio_track = static_cast<OTIO::Track*>(track.value);
+      number_of_clips += otio_track->children().size();
+    }
+  }
+
+  // Dialog has to be called from the main thread so we pass the list of sequences here.
+  QMetaObject::invokeMethod(Core::instance(),
+                            "DialogImportOTIOShow",
+                            Qt::BlockingQueuedConnection,
+                            Q_ARG(QList<Sequence*>,timeline_sequnce_map.values()));
```

Comment:
Should return a bool from DialogImportOTIOShow if `QDialog::exec() == QDialog::accepted` and cancel loading here if false. I found clicking cancel on the dialog would still load the project, which feels a little odd.